cmd/clusterctl/test/run-e2e.sh

```diff
@@ -0,0 +1,42 @@
+#!/bin/bash
+
+# Copyright 2020 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+REPO_ROOT=$(dirname "${BASH_SOURCE[0]}")/../../..
+cd "${REPO_ROOT}" || exit 1
+REPO_ROOT_ABS=${PWD}
+ARTIFACTS="${ARTIFACTS:-${PWD}/_artifacts}"
+mkdir -p "$ARTIFACTS/logs/"
+
+cd "${REPO_ROOT_ABS}/test/infrastructure/docker"
+export REGISTRY=gcr.io/"$(gcloud config get-value project)"
+export DOCKER_MANAGER_IMAGE=docker-provider-manager
+export TAG=dev
+export ARCH=amd64
+# Update the images used in the e2e tests.
+# This will load the capd image into the kind mgmt cluster.
+export MANAGER_IMAGE=${REGISTRY}/${DOCKER_MANAGER_IMAGE}-${ARCH}:${TAG}
+export PULL_POLICY=IfNotPresent
+if [ -z "${SKIP_DOCKER_BUILD-}" ]; then
+	CONTROLLER_IMG=${REGISTRY}/${DOCKER_MANAGER_IMAGE} make docker-build
+fi
+GINKGO_FOCUS=${GINKGO_FOCUS:-""}
```

Comment:
The script is moving back and forth from building the docker image and setting up what is required for running ginko (e.g. ARTIFACT folder).
WDYT about refactor a little bit in order to get two separate sections with a comment before?